### PR TITLE
Feat(MicroSplit API): MicroSplit and HDN configurations and factories

### PR DESCRIPTION
**Pull request by CatEek**
## Disclaimer

- [ ] I am an AI agent.
- [x] I have used AI and I thoroughly reviewed every line.
- [ ] I have not used AI extensively.

## Description

> [!NOTE]  
> **tldr**: Add NG `create_*_config` / `create_advanced_*_config` factories for HDN and MicroSplit, returning validated `Configuration` objects in the same style as CARE/N2N/N2V.

### Background - why do we need this PR?

Bring HDN and MicroSplit configuration up to current standard

### Overview - what changed?

- Two new configuration classes (`HDNConfiguration`, `MicroSplitConfiguration`)
- Two new factory modules mirroring the `care_n2n_factory.py` 
- Tests for factory behavior, checkpoint presets, and algorithm-specific defaults.

### Implementation - how did you implement the changes?

Each factory builds a complete, validated configuration object from keyword arguments:

1. **Algorithm config** — `LVAELossConfig`, `LVAEConfig`, and the algorithm-specific Pydantic model (`HDNAlgorithm` / `MicroSplitAlgorithm`).
2. **Data config** — `create_ng_data_configuration` for HDN; same base config wrapped into `MicroSplitDataConfig` for MicroSplit (multiscale, padding, alpha ranges, uncorrelated channel probability).
3. **Training config** — `TrainingConfig` assembled directly in each factory (matching `care_n2n_factory.py`), with algorithm-appropriate checkpoint/early-stopping presets.

Example usage:

```python
from careamics.config import create_hdn_config, create_microsplit_config

hdn_cfg = create_hdn_config(
    experiment_name="hdn-demo",
    data_type="tiff",
    axes="SYX",
    patch_size=[64, 64],
    batch_size=8,
)

ms_cfg = create_microsplit_config(
    experiment_name="ms-demo",
    data_type="tiff",
    axes="SYX",
    patch_size=[64, 64],
    batch_size=8,
    multiscale_count=3,
)
```

### New features or files

- `HDNConfiguration` added to `config/hdn_configuration.py`
- `MicroSplitConfiguration` added to `config/microsplit_configuration.py`
- `create_hdn_config` / `create_advanced_hdn_config` added to `config/factories/hdn_factory.py`
- `create_microsplit_config` / `create_advanced_microsplit_config` added to `config/factories/microsplit_factory.py`
- 8 tests added to `tests/config/factories/test_hdn_factory.py`
- 6 tests added to `tests/config/factories/test_microsplit_factory.py`

### Modified features or files

- `config/__init__.py` — export the 4 new factory functions
- `config/factories/__init__.py` — import/export the 4 new factory functions
- `compat/config/configuration_factories.py` — deprecation notes on `create_hdn_configuration` and `create_microsplit_configuration`

## How has this been tested?
with tests

## Notes
LVAELossConfig might need to be split/modified for HDN

## Related Issues
#979 
#849 
#840 
---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

**Comment by jdeschamps** [comment]
I have not looked at it yet, but you mention HDN. However, it seems to me that @veegalinova implemented HDN configuration factories in https://github.com/CAREamics/careamics/pull/994. Are those the same? Did you take these and modify them?

[EDIT]: #994 only created a convenience function for HDN algorithm config in order to perform tests. So this PR should be merged first.

File: src/careamics/compat/config/configuration_factories.py
Comment (jdeschamps) [inline src/careamics/compat/config/configuration_factories.py:1]: As we discussed, all `careamics.compat` should be removed.

File: src/careamics/config/factories/hdn_factory.py
Comment (jdeschamps) [inline src/careamics/config/factories/hdn_factory.py:1]: How do these compare to the factory in https://github.com/CAREamics/careamics/pull/994?

File: src/careamics/config/factories/microsplit_factory.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    TrainingConfig,
+)
+from careamics.config.losses.loss_config import LVAELossConfig
+from careamics.config.microsplit_configuration import MicroSplitConfiguration
+from careamics.config.noise_model.noise_model_config import MultiChannelNMConfig
+
+from .data_factory import create_ng_data_configuration, list_spatial_augmentations
+from .training_factory import update_trainer_params
+
+NonLinearity = Literal["None", "Sigmoid", "Softmax", "Tanh", "ReLU", "LeakyReLU", "ELU"]
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: Do we need these? I was under the impression that the same non linearity has been consistently used in MSplit.

File: src/careamics/config/factories/microsplit_factory.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    experiment_name: str,
+    data_type: Literal["array", "tiff", "zarr", "czi", "custom"],
+    axes: str,
+    patch_size: Sequence[int],
+    batch_size: int,
+    num_epochs: int = 100,
+    num_steps: int | None = None,
+    augmentations: Sequence[Literal["x_flip", "y_flip", "rotate_90"]] | None = None,
+    n_val_patches: int = 8,
+    output_channels: int = 1,
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: Does a default value for `output_channels` equal to `1` make sense for MSplit? Shouldn't it be specified by the users?

File: src/careamics/config/factories/microsplit_factory.py [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+    multiscale_count: int = 3,
+    noise_model: MultiChannelNMConfig | None = None,
+) -> MicroSplitConfiguration:
+    """Create a configuration for training MicroSplit.
+
+    A noise model is required for denoiSplit training; if omitted while
+    `denoisplit_weight > 0`, a warning is raised and the noise model must be attached
+    before training.
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: That seems unrelated to this function.

```suggestion
```

File: src/careamics/config/factories/microsplit_factory.py [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+    output_channels: int = 1,
+    z_dims: Sequence[int] = (128, 128),
+    encoder_n_filters: int = 32,
+    decoder_n_filters: int = 32,
+    encoder_dropout: float = 0.1,
+    decoder_dropout: float = 0.1,
+    nonlinearity: NonLinearity = "ELU",
+    analytical_kl: bool = False,
+    predict_logvar: bool = True,
+    logvar_lowerbound: float | None = -5.0,
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: why not fold these into `model_parameters`?

File: src/careamics/config/factories/microsplit_factory.py [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+    # loss parameters
+    reconstruction_weight: float = 1.0,
+    kl_weight: float = 1.0,
+    musplit_weight: float = 0.1,
+    denoisplit_weight: float = 0.9,
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py:105]: We might want to pass this as a LossConfig already. We have done this for the patching and maybe it is easier than dragging a billion parameter.

I am not entirely sold to the idea yet.

File: src/careamics/config/factories/microsplit_factory.py [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+    # algorithm parameters
+    noise_model: MultiChannelNMConfig | None = None,
+    mmse_count: int = 10,
+    # lightning parameters
+    num_workers: int = -1,
+    trainer_params: dict | None = None,
+    optimizer: Literal["Adam", "Adamax", "SGD"] = "Adamax",
+    lr: float = 1e-3,
+    lr_scheduler_patience: int = 30,
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: why are these explicit and not folded into `optimizer_parameters` and `lr_scheduler_parameters`? are they changed often for MSplit?

File: src/careamics/config/factories/microsplit_factory.py [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+    kl_weight: float = 1.0,
+    musplit_weight: float = 0.1,
+    denoisplit_weight: float = 0.9,
+    # algorithm parameters
+    noise_model: MultiChannelNMConfig | None = None,
+    mmse_count: int = 10,
+    # lightning parameters
+    num_workers: int = -1,
+    trainer_params: dict | None = None,
+    optimizer: Literal["Adam", "Adamax", "SGD"] = "Adamax",
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py:112]: where are `optimizer_parameters`?

File: src/careamics/config/factories/microsplit_factory.py [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+                "min_lr": 1e-12,
+            },
+        ),
+    )
+
+    norm_config = {"name": normalization}
+    if normalization_params is not None:
+        norm_config.update(normalization_params)
+
+    augs: list[XYFlipConfig | XYRandomRotate90Config] | None = None
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py:273]: Adding a TODO here because this will be refactored somewhere else once https://github.com/CAREamics/careamics/pull/1005 is merged.

```suggestion

    # TODO refactor when #1005 will be merged
    augs: list[XYFlipConfig | XYRandomRotate90Config] | None = None
```

File: src/careamics/config/factories/microsplit_factory.py [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+            checkpoint_params
+            if checkpoint_params is not None
+            else asdict(SupervisedCheckpointing())
+        ),
+        early_stopping_params=(
+            early_stopping_params
+            if early_stopping_params is not None
+            else {"monitor": "val_loss", "mode": "min"}
+        ),
+    )
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: why not using `create_training_configuration` and leveraging the default checkpointing and early stropping from the training configuration (by adding msplit next to care)?

File: src/careamics/config/hdn_configuration.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    @model_validator(mode="after")
+    def validate_patch_against_model(self: Self) -> Self:
+        """Skip UNet model-constraint validation for LVAE models.
+
+        Returns
+        -------
+        Self
+            Validated configuration.
+        """
+        return self
```
Comment (jdeschamps) [inline src/careamics/config/hdn_configuration.py:31]:
```suggestion
    # TODO remove once LVAE model constraints will be implemented
    @model_validator(mode="after")
    def validate_patch_against_model(self: Self) -> Self:
        """Skip UNet model-constraint validation for LVAE models.

        Returns
        -------
        Self
            Validated configuration.
        """
        return self
```

File: src/careamics/config/factories/microsplit_factory.py [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+    data_type: Literal["array", "tiff", "zarr", "czi", "custom"],
+    axes: str,
+    patch_size: Sequence[int],
+    batch_size: int,
+    num_epochs: int = 100,
+    num_steps: int | None = None,
+    augmentations: Sequence[Literal["x_flip", "y_flip", "rotate_90"]] | None = None,
+    n_val_patches: int = 8,
+    # advanced data parameters
+    channels: Sequence[int] | None = None,
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py:89]: I don't think we should make `channels` a parameter for MSplit given how painful assembling the data is. That means that a validation should be there in the configuration to enforce `channels is None`

File: src/careamics/config/hdn_configuration.py [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+            Validated configuration.
+        """
+        n_in = (
+            len(self.data_config.channels)
+            if self.data_config.channels is not None
+            else 1
+        )
+        n_out = self.algorithm_config.model.output_channels
+        self.data_config.normalization.validate_size(n_in, n_out)
+        return self
```
Comment (jdeschamps) [inline src/careamics/config/hdn_configuration.py:62]:
```suggestion
    # TODO remove once LVAE model constraints have been implemented
    @model_validator(mode="after")
    def validate_channels_against_inputs(self: Self) -> Self:
        """Skip UNet channel-constraint validation for LVAE models.

        Returns
        -------
        Self
            Validated configuration.
        """
        return self
        
    # TODO remove once LVAE model constraints have been implemented
    @model_validator(mode="after")
    def validate_norm_against_channels(self: Self) -> Self:
        """Validate that normalization sizes match the LVAE channels.

        Returns
        -------
        Self
            Validated configuration.
        """
        n_in = (
            len(self.data_config.channels)
            if self.data_config.channels is not None
            else 1
        )
        n_out = self.algorithm_config.model.output_channels
        self.data_config.normalization.validate_size(n_in, n_out)
        return self
```

File: src/careamics/config/microsplit_configuration.py [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+        n_in = (
+            len(self.data_config.channels)
+            if self.data_config.channels is not None
+            else 1
+        )
+        n_out = self.algorithm_config.model.output_channels
+        self.data_config.normalization.validate_size(n_in, n_out)
+        return self
+
+    @model_validator(mode="after")
```
Comment (jdeschamps) [inline src/careamics/config/microsplit_configuration.py]: Not a fan of empty validator. The LVAE config should implement `self.algorithm_config.model.uses_batch_norm()` that always returns `False`. Then the validator stays unchanged in the parent class.

File: src/careamics/config/hdn_configuration.py [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+    @model_validator(mode="after")
+    def warn_batch_norm(self: Self) -> Self:
```
Comment (jdeschamps) [inline src/careamics/config/hdn_configuration.py]: See comment in the MSplit configuration for the same validator.

File: src/careamics/config/microsplit_configuration.py
Comment (jdeschamps) [inline src/careamics/config/microsplit_configuration.py:1]: See comments adding TODOs in HDN config

File: tests/config/factories/test_microsplit_factory.py [hunk 15]
```diff
@@ -0,0 +1,101 @@
+import pytest
+
+from careamics.config.algorithms import MicroSplitAlgorithm
+from careamics.config.data import MicroSplitDataConfig
+from careamics.config.factories import (
+    create_advanced_microsplit_config,
+    create_microsplit_config,
+)
+from careamics.config.microsplit_configuration import MicroSplitConfiguration
+
```
Comment (jdeschamps) [inline tests/config/factories/test_microsplit_factory.py:10]:
```suggestion
# TODO refactor similarly to https://github.com/CAREamics/careamics/pull/1005 once it is merged
```

File: tests/config/factories/test_hdn_factory.py [hunk 16]
```diff
@@ -0,0 +1,111 @@
+import pytest
+
+from careamics.config.algorithms import HDNAlgorithm
+from careamics.config.factories import (
+    create_advanced_hdn_config,
+    create_hdn_config,
+)
+from careamics.config.hdn_configuration import HDNConfiguration
+
```
Comment (jdeschamps) [inline tests/config/factories/test_hdn_factory.py:9]:
```suggestion
# TODO refactor similarly to https://github.com/CAREamics/careamics/pull/1005 once it is merged
```

**Review by jdeschamps — CHANGES_REQUESTED** [review]

File: src/careamics/compat/config/configuration_factories.py
Comment (jdeschamps) [reply src/careamics/compat/config/configuration_factories.py:1]: It is now a conflict with `dev/microsplit_api`

File: src/careamics/config/factories/microsplit_factory.py [hunk 17]
```diff
@@ -0,0 +1,352 @@
...
+    alpha_ranges: Sequence[tuple[float, float]] | None = None,
+    uncorrelated_channel_prob: float = 0.0,
+    # model parameters
+    output_channels: int = 1,
+    z_dims: Sequence[int] = (128, 128),
+    encoder_n_filters: int = 32,
+    decoder_n_filters: int = 32,
+    encoder_dropout: float = 0.1,
+    decoder_dropout: float = 0.1,
+    nonlinearity: NonLinearity = "ELU",
```
Comment (jdeschamps) [inline src/careamics/config/factories/microsplit_factory.py]: do we ever change that?

File: src/careamics/config/factories/hdn_factory.py
Comment (jdeschamps) [reply src/careamics/config/factories/hdn_factory.py:1]: Ok, I just had a look and #994 takes a shortcut of creating just the algorithm config in order to test HDN. This version should have priority.

File: src/careamics/config/factories/microsplit_factory.py (hunk 11 above)
Comment (melisande-c) [reply src/careamics/config/factories/microsplit_factory.py:89]: I have already implemented the channels option in the dataset microsplit patch constructors:

https://github.com/CAREamics/careamics/blob/6d28bc7d9b94d9b339cc2f5a0bcd0c27b74d414c/src/careamics/dataset/patch_constructor/microsplit_patch_constructors.py#L60

Comment (jdeschamps) [reply src/careamics/config/factories/microsplit_factory.py:89]: Ok, out of scope then.

That being said, I am changing my mind a bit: when using already acquired datasets, this feature is actually useful. I was only considering data acquired specifically for MSplit (and even then, can be useful).

Going further, a feature that might be useful for MSplit is `target_channels`. I can imagine (e.g. Diya's project) where MSplit may be trained with mode 3 but with a subset of the overall acquired channels. The study might require the acquisition of 5 channels but only 3 of them are mixed (e.g. optimization of MSplit performances, some structures being too similar). There it would be quite handy to not touch the multi-channel data but only specify `target_channels`.

File: src/careamics/config/factories/microsplit_factory.py (hunk 17 above)
Comment (CatEek) [reply src/careamics/config/factories/microsplit_factory.py]: Dropouts unlikely, decoder filters, might happen

Comment (jdeschamps) [reply src/careamics/config/factories/microsplit_factory.py]: And the non linearity?

File: src/careamics/config/factories/microsplit_factory.py (hunk 1 above)
Comment (CatEek) [reply src/careamics/config/factories/microsplit_factory.py]: Removed

File: src/careamics/config/factories/microsplit_factory.py (hunk 17 above)
Comment (CatEek) [reply src/careamics/config/factories/microsplit_factory.py]: removed the param

File: src/careamics/config/factories/microsplit_factory.py (hunk 5 above)
Comment (CatEek) [reply src/careamics/config/factories/microsplit_factory.py:105]: I created an issue, I'd investigate it and do a separatte PR

File: src/careamics/config/factories/microsplit_factory.py (hunk 7 above)
Comment (CatEek) [reply src/careamics/config/factories/microsplit_factory.py:112]: fixed

File: src/careamics/config/factories/microsplit_factory.py (hunk 6 above)
Comment (CatEek) [reply src/careamics/config/factories/microsplit_factory.py]: done

File: src/careamics/config/microsplit_configuration.py (hunk 13 above)
Comment (CatEek) [reply src/careamics/config/microsplit_configuration.py]: fixed

File: src/careamics/compat/config/configuration_factories.py
Comment (CatEek) [reply src/careamics/compat/config/configuration_factories.py:1]: removed
